execution/executor/src/block_executor.rs

```diff
@@ -220,19 +220,35 @@ where
         sync_commit: bool,
     ) -> Result<(), Error> {
         let _timer = APTOS_EXECUTOR_COMMIT_BLOCKS_SECONDS.start_timer();
+
+        // Ensure the block ids are not empty
+        if block_ids.is_empty() {
+            return Err(anyhow::anyhow!("Cannot commit 0 blocks!").into());
+        }
+
+        // Check for any potential retries
         let committed_block = self.block_tree.root_block();
         if committed_block.num_persisted_transactions()
             == ledger_info_with_sigs.ledger_info().version() + 1
         {
-            // a retry
```

Comment:
why?

Reply:
Comment was moved above: `// Check for any potential retries`. Let me know if you want both.